mood-analysis.rb

```diff
@@ -19,6 +24,34 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def happy_days(text)
+  smile_count = 0
+  day_count = 0
+  text.each do |line|
+    day_count += 1
+    if analyze_mood(line) == ":-)"
+      smile_count += 1
+    end
+    if smile_count == 3
+      break
+    end
+  end
+  if smile_count == 3
+    return "It took #{day_count} days for happiness to occur."
+  else
+    return "Sorry, no happy days..."
+  end
+end
+
+def overall_mood(text)
+moods_array = []
+  text.each do |line|
+    moods_array << analyze_mood(line)
+  end
+overall_mood = moods_array.max_by{|mood| moods_array.count(mood)}
+return "The most common mood is #{overall_mood}"
+end
+
```

Comment:
Also, watch your formatting here. Everything in def overall_mood should be tabbed over one.